ext/src/ruby_api/wasi_deterministic_ctx_builder.rs

```diff
@@ -0,0 +1,202 @@
+use super::root;
```

Comment:
This and `wasi_ctx_builder` have many similarities. When refactoring see if there is a way to merge these.